addons/sale/models/sale_order_line.py

```diff
@@ -391,13 +396,17 @@ def _get_price_rule_id(self, **product_context):
 
         if pricelist_rule:
             price = pricelist_rule._compute_price(
-                product, qty, self.product_uom, order_date)
+                product, qty, self.currency_id, self.product_uom, order_date)
         else:
             # fall back on Sales Price if no rule is found
-            price = product.price_compute('list_price', uom=self.product_uom, date=order_date)[product.id]
+            price = product.price_compute(
+                'list_price', uom=self.product_uom, currency=self.currency_id, date=order_date
+            )[product.id]
```

Comment:
Keep this part, use a `src_currency` var and unindent the if below (we do not want to round the amount, which is done by default in `price_compute`) ?
```python
if pricelist_rule:
     src_currency = pricelist_rule.currency_id
else:
    src_currency = product.currency_id

if src_currency != self.currency_id:
    ...
```